### DISCUSSION TITLE
Template .gitignore with quarto create-project

### DISCUSSION THREAD

**Discussion by dmbates** [quarto-dev/quarto-cli#575] [Q&A] [answered]
Is there a template .gitignore file for a quarto project.  I am interested in what files in the `_book` directory should be kept in a git repository and which can be safely omitted because they will be regenerated.

If there is such a file, could it be added to the files installed by `quarto create-project`?

**Comment by jjallaire** [comment, marked as answer]
Currently we don't create the `.gitignore` file upon `create-project` but rather we do it during `quarto render` if we see that the project is within a git repo (the idea being not to auto-create files not strictly required).

Note that this logic currently resolves to the following `.gitignore` for all projects:

```
/.quarto/
```

Additionally, if we detect a Python virtual environment in `env/` we add that as well:

```
/.quarto/
/env/
```

Again, all of this occurs during `quarto render` rather than `quarto create-project`.

**Reply by haxfn** [reply]
Where is this documented?

**Reply by mcanouil** [reply]
@vishalpaudel this is a more than 3-years old thread. Please open a new discussion eventually referencing this one and add context to your question. Thank you.
